packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -71,19 +117,93 @@ class TimeInput extends React.Component {
     this.handleFocus = this.handleFocus.bind(this);
     this.handleHourBlur = this.handleHourBlur.bind(this);
     this.handleMinuteBlur = this.handleMinuteBlur.bind(this);
+    this.handleMeridiemChange = this.handleMeridiemChange.bind(this);
+    this.handleMeridiemInputKeyDown = this.handleMeridiemInputKeyDown.bind(this);
+    this.meridiemFocus = this.meridiemFocus.bind(this);
+    this.meridiemBlur = this.meridiemBlur.bind(this);
+    this.handleMeridiemButtonChange = this.handleMeridiemButtonChange.bind(this);
+
+    let hour = TimeUtil.splitHour(value);
+    let meridiem = null;
+
+    if (props.variant === FORMAT_12_HOUR) {
+      this.anteMeridiem = context.intl.formatMessage({ id: 'Terra.timeInput.am' });
+      this.postMeridiem = context.intl.formatMessage({ id: 'Terra.timeInput.pm' });
+
+      if (hour) {
+        const parsedHour = this.parseTimeTwelveHourTime(hour, true);
+        hour = parsedHour.hourString;
+        meridiem = parsedHour.meridiem;
+      } else {
+        meridiem = this.anteMeridiem;
+      }
+    }
+
+    this.state = {
+      hour,
+      minute: TimeUtil.splitMinute(value),
+      isFocused: false,
+      meridiem,
+      meridiemFocused: false,
+    };
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.value === this.props.value) {
+    if (
+      nextProps.value === this.props.value &&
+      nextProps.variant === this.props.variant
+    ) {
       return;
     }
 
+    let hour = TimeUtil.splitHour(nextProps.value);
```

Comment:
lines 159-171 are almost 1-to-1 with the 12-hour logic in the constructor. Maybe we should break this out to a reusable function in Time Utils?

Reply:
The hard part is that one of them is the constructor, which doesn't have the props and context set already, and the other is the component update function, which deals with two sets of props and also having the context attached to the actual component. I gave it a refactoring attempt, but felt that if involved too much passing around of parameters and didn't provide a lot of refactoring value. I'll look at more ideas, but I think it reads better as it is atm.

Reply:
Set the default to undefined right here. https://github.com/cerner/terra-core/pull/1027/commits/d0d2cadb3ad5a2456ca691b9f7234ac2ac14396d